### PR TITLE
F asep assignments

### PR DESCRIPTION
**Pull request by asepnur**
check it first bro

### CODE REVIEW

File: src/module/assignment/model.go [hunk 1]
```diff
@@ -1,13 +1,74 @@
...
+}
+
+// FileAssignment struct ...
+type FileAssignment struct {
+	Assignment Assignment
+}
+
+// GradeParameter struct ...
+type GradeParameter struct {
+	Type         string  `json:"type"`
```
Comment (melodiez14) [inline src/module/assignment/model.go]: put json in webhandler struct

File: src/module/assignment/model.go [hunk 2]
```diff
@@ -1,13 +1,74 @@
...
+//DetailAssignment struct ...
+type DetailAssignment struct {
+	Assignment     Assignment
+	File           File
+	GradeParameter GradeParameter
+}
+
+// DetailUploadedAssignment struct ...
+type DetailUploadedAssignment struct {
+	ScheudleID            int64          `json:"schdule_id"`
```
Comment (melodiez14) [inline src/module/assignment/model.go]: typo and put json in web handler

File: src/module/assignment/assignment.go [hunk 3]
```diff
@@ -3,10 +3,14 @@ package assignment
...
 	"fmt"
+	"strconv"
+	"time"
 
+	"github.com/jmoiron/sqlx"
 	"github.com/melodiez14/meiko/src/util/conn"
 )
 
+// GetByCourseID func ...
 func GetByCourseID(courseID int64) ([]Assignment, error) {
```
Comment (melodiez14) [inline src/module/assignment/assignment.go]: Change SelectByCourseID
Select returns slice, get return single object

File: src/module/assignment/assignment.go [hunk 4]
```diff
@@ -17,6 +21,7 @@ func GetByCourseID(courseID int64) ([]Assignment, error) {
 	return assignments, nil
 }
 
+// GetIncompleteByUserID func ...
 func GetIncompleteByUserID(userID int64) ([]Assignment, error) {
```
Comment (melodiez14) [inline src/module/assignment/assignment.go]: Same as before

File: src/module/assignment/assignment.go [hunk 5]
```diff
@@ -27,6 +32,7 @@ func GetIncompleteByUserID(userID int64) ([]Assignment, error) {
 	return assignments, nil
 }
 
+// GetCompleteByUserID func ...
 func GetCompleteByUserID(userID int64) ([]int64, error) {
```
Comment (melodiez14) [inline src/module/assignment/assignment.go]: same as before

File: src/module/assignment/assignment.go [hunk 6]
```diff
@@ -54,3 +61,507 @@ func IsExistByGradeParameterID(gpID int64) bool {
...
+	id, err := result.LastInsertId()
+	if err != nil {
+		return "", fmt.Errorf("Error get LastIsertedId")
+	}
+	ID := strconv.FormatInt(id, 10)
+	return ID, nil
+}
+
+// Update func ...
+func Update(GradeParameters, id int64, Name, Status, DueDate string, Description sql.NullString, tx *sqlx.Tx) error {
```
Comment (melodiez14) [inline src/module/assignment/assignment.go]: change the character with private (lower case for the first character)

File: src/module/assignment/assignment.go [hunk 7]
```diff
@@ -54,3 +61,507 @@ func IsExistByGradeParameterID(gpID int64) bool {
...
+	}
+	rows, err := result.RowsAffected()
+	if rows == 0 {
+		return fmt.Errorf("No rows affected")
+	}
+	return nil
+}
+
+// IsFileIDExist func ...
+func IsFileIDExist(ID string) bool {
```
Comment (melodiez14) [inline src/module/assignment/assignment.go]: good

File: src/module/assignment/assignment.go [hunk 8]
```diff
@@ -54,3 +61,507 @@ func IsExistByGradeParameterID(gpID int64) bool {
...
+			`, ID)
+	err := conn.DB.Get(&x, query)
+	if err == sql.ErrNoRows {
+		return false
+	}
+	return true
+}
+
+// SelectByPage func ...
+func SelectByPage(limit, offset uint16) ([]FileAssignment, error) {
```
Comment (melodiez14) [inline src/module/assignment/assignment.go]: use []Assignment

File: src/module/assignment/assignment.go [hunk 9]
```diff
@@ -54,3 +61,507 @@ func IsExistByGradeParameterID(gpID int64) bool {
...
+				asg.grade_parameters_id,
+				asg.name,
+				asg.description,
+				asg.status,
+				asg.due_date
+			FROM
+				assignments asg
+			LIMIT %d OFFSET %d;`, limit, offset)
+
+	rows, err := conn.DB.Queryx(query)
```
Comment (melodiez14) [inline src/module/assignment/assignment.go]: why dont use conn.DB.Select(query) ?

File: src/module/assignment/assignment.go (hunk 6 above)
Comment (asepnur) [reply src/module/assignment/assignment.go]: done

File: src/module/assignment/assignment.go (hunk 3 above)
Comment (asepnur) [reply src/module/assignment/assignment.go]: done

File: src/module/assignment/assignment.go (hunk 4 above)
Comment (asepnur) [reply src/module/assignment/assignment.go]: done

File: src/module/assignment/assignment.go (hunk 5 above)
Comment (asepnur) [reply src/module/assignment/assignment.go]: done

File: src/module/assignment/assignment.go (hunk 8 above)
Comment (asepnur) [reply src/module/assignment/assignment.go]: done

File: src/module/assignment/assignment.go (hunk 9 above)
Comment (asepnur) [reply src/module/assignment/assignment.go]: dari documentasi

File: src/module/assignment/model.go (hunk 1 above)
Comment (asepnur) [reply src/module/assignment/model.go]: done

File: src/module/assignment/model.go (hunk 2 above)
Comment (asepnur) [reply src/module/assignment/model.go]: done

**Pull request merged**